client/web/src/site-admin/SiteAdminBackgroundJobsPage.tsx

```diff
@@ -0,0 +1,483 @@
+        routine.type !== BackgroundRoutineType.DB_BACKED &&
+        (!lastRecentRunDate ||
+            lastRecentRunDate.getTime() + routine.intervalMs + routine.stats.maxDurationMs + 5000 <= Date.now())
+
+    const tooltip = isStopped
+        ? `This routine is currently stopped.
+Started at ${format(new Date(latestStartDateString), 'yyyy-MM-dd HH:mm:ss')},
+Stopped at: ${format(new Date(earliestStopDateString), 'yyyy-MM-dd HH:mm:ss')}`
+        : isUnseenInAWhile
+        ? lastRecentRunDate
+            ? `This routine has not been seen in a while. It should've run at ${format(
+                  new Date(lastRecentRunDate.getTime() + (routine.intervalMs || 0)),
+                  'yyyy-MM-dd HH:mm:ss'
+              )}.`
+            : 'This routine was started but it has never been seen running.'
+        : `This routine is currently started.${
+              lastRecentRunDate ? `\nLast seen running at ${format(lastRecentRunDate, 'yyyy-MM-dd HH:mm:ss')}.` : ''
+          }`
+
+    return isStopped || isUnseenInAWhile ? (
+        <Tooltip content={tooltip}>
+            <Icon aria-label="stopped" svgPath={mdiClose} className="text-danger" />
+        </Tooltip>
+    ) : (
+        <Tooltip content={tooltip}>
+            <Icon aria-label="started" svgPath={mdiCheck} className="text-success" />
+        </Tooltip>
+    )
+}
+
+function isRoutineProblematic(routine: BackgroundJob['routines'][0]): boolean {
+    return (
+        routine.stats.errorCount > 0 ||
+        routine.recentRuns.some(
+            run => run.errorMessage || categorizeRunDuration(run.durationMs, routine.intervalMs) !== 'short'
+        ) ||
+        categorizeRunDuration(routine.stats.minDurationMs, routine.intervalMs) !== 'short' ||
+        categorizeRunDuration(routine.stats.avgDurationMs, routine.intervalMs) !== 'short' ||
+        categorizeRunDuration(routine.stats.maxDurationMs, routine.intervalMs) !== 'short'
+    )
+}
+
+// Contains some magic numbers
+function categorizeRunDuration(durationMs: number, routineIntervalMs: number | null): RunLengthCategory {
+    if (!routineIntervalMs) {
+        return durationMs > 5000 ? 'long' : 'short'
+    }
+    if (durationMs > routineIntervalMs * 0.7) {
+        return 'dangerous'
+    }
+    // Uses both a relative and an absolute filter:
+    // If the run is more than 10% longer than the interval, it's long, except for intervals of 1s or less where 500ms+ is long.
+    // Also, any run longer than 5s is long.
+    if (durationMs > routineIntervalMs * (durationMs > 1000 ? 0.1 : 0.5) || durationMs > 5000) {
+        return 'long'
+    }
+    return 'short'
+}
+
+function getRunDurationTextClass(durationMs: number, routineIntervalMs: number | null): string {
+    const category = categorizeRunDuration(durationMs, routineIntervalMs)
+    if (category === 'dangerous') {
+        return 'text-danger'
+    }
+
+    if (category === 'long') {
+        return 'text-warning'
+    }
+
+    return 'text-success'
```

Comment:
Switch-a-roo? 😄 I feel they communicate mutual exclusivity a bit more clearly.

```suggestion
    switch (category) {
       case 'dangerous':
          return 'text-danger'
       case 'long':
          return 'text-warning'
       default:
          return 'text-success'
     }
```